test/SFSmartSyncTestSuite.js

```diff
@@ -2871,6 +2942,77 @@ SmartSyncTestSuite.prototype.testSyncUpLocallyDeleted = function() {
 };
 
 /** 
+ * TEST smartsyncplugin sync up with locally deleted records and merge mode leave-if-changed
+ */
+SmartSyncTestSuite.prototype.testSyncUpLocallyDeletedWithNoOverwrite = function() {
+    console.log("# In SmartSyncTestSuite.testSyncUpLocallyDeletedWithNoOverwrite");
+    var self = this;
+    var idToName = {};
+    var idToUpdatedName = {};
+    var deletedRecords;
+    var options = {fieldlist: ["Name"], mergeMode: Force.MERGE_MODE_DOWNLOAD.LEAVE_IF_CHANGED};
+    var soupName = "testSyncUpLocallyDeletedWithNoOverwrite";
+    var cache;
+
+    Force.smartstoreClient.removeSoup(soupName)
+        .then(function() {
+            console.log("## Initialization of StoreCache's");
+            cache = new Force.StoreCache(soupName, [ {path:"Name", type:"string"} ]);
+            return $.when(cache.init());
+        })
+        .then(function() { 
+            console.log("## Direct creation against server");
+            return createRecords(idToName, "testSyncUpLocallyDeletedWithNoOverwrite", 3);
+        })
+        .then(function() {
+            console.log("## Calling sync down");
+            return self.trySyncDown(cache, soupName, idToName, Force.MERGE_MODE_DOWNLOAD.LEAVE_IF_CHANGED);
+        })
+        .then(function() {
+            console.log("## Deleted local records");
+            deletedRecords = [];
+            _.each(_.keys(idToName), function(id) {
+                deletedRecords.push({Id:id, __locally_deleted__:true});
+            });
+            return cache.saveAll(deletedRecords);
+        })
+        .then(function() {
```

Comment:
Same as the other ```syncUp``` test, except this addition here, to edit the server record before attempting to sync up again.